Feat/font_logo_header_menu

mikemangas opened the pull request:
Added following basic styling:

1. Font Family
2. Defined font-size in root for relevant content
3. Added Icon
4. Added style for header
5. Added toggle function for header

namirsab commented:
@mikemangas your app is somehow not deploying. Please fix that before

namirsab commented on client/src/App.css:
```diff
@@ -3,6 +3,7 @@
 @import url("./styles/_colors.css");
 @import url("./styles/Submitform.css");
 @import url("./styles/Home.css");
+@import url("./styles/Header.css");
```
Actually, with `cra` and create react app is much better to load each css file in each component `js` file.

mikemangas replied:
ok thx

namirsab commented on client/src/assets/Images/Spielplatzfinder-Dateien/bundle.js:
```diff
@@ -0,0 +1,963 @@
+/******/ (function(modules) { // webpackBootstrap
```
hm, what is this? This doesn't look like an image

mikemangas replied:
no idea how it appeared ❌

namirsab commented on client/src/assets/Images/Spielplatzfinder-Dateien/leaflet.css:
```diff
@@ -0,0 +1,640 @@
+/* required styles */
```
Also this, this doesn't look like an image and i'm not sure if you need to have it locally?

mikemangas replied:
trying to fix asap.

namirsab commented on client/src/assets/Images/Spielplatzfinder-Dateien/main.js:
```diff
@@ -0,0 +1,2485 @@
+(this["webpackJsonpclient"] = this["webpackJsonpclient"] || []).push([["main"],{
```
hm 🤨

mikemangas replied:
i didnt create that

namirsab replied:
This looks like you've built your app, and moved the build folder contents inside here or something like this

namirsab reviewed (changes requested):
@mikemangas this is just a partial review. I have the feeling you've pushed many files by accident, that shouldn't have been pushed.

Please take a look to all the files you have under your `Images/` folder

mikemangas commented:
> @mikemangas your app is somehow not deploying. Please fix that before

trying to fix asap

mikemangas commented:
> @mikemangas your app is somehow not deploying. Please fix that before

i fixed the deployment:

problem: i previously had fixed a typo in the file name of the css.. but it somehow didnt update this fixing of the name...
solution: i moved the file to another folder. did git add... and brought it back to the correct folder... and then... it just took the ammendment in the file name!

namirsab commented on client/src/index.css:
```diff
@@ -1,7 +1,21 @@
 *,
 ::before,
 ::after {
   box-sizing: content-box;
```
`content-box` is default, you probably want `border-box` which is the better way. 
https://css-tricks.com/box-sizing/

namirsab commented on client/src/index.css:
```diff
@@ -1,7 +1,21 @@
+@import url("./styles/_base.css");
+@import url("./styles/_colors.css");
+@import url("./styles/_fonts.css");
```
You could import all these files in your index.js

namirsab commented on client/src/components/Header.css:
```diff
@@ -0,0 +1,46 @@
+.Header {
+  display: grid;
+  grid-template-columns: [logo] 60px [navigation] auto;
+  padding: 20px;
```
Paddings should use `em` or `rem` (em if you want the padding to depend on the font-size of the element, rem on the root font-size)

namirsab commented on client/src/components/Header.js:
```diff
@@ -1,20 +1,57 @@
+    "Header__navigation--toggle--off"
+  );
+
   if (localStorage.getItem("userId") === null) {
     return localStorage.setItem("userId", JSON.stringify(uuidv4()));
   }
```
Whay are you returning in here? This would make the component not render anything, or worse, it can crash! Take a look

namirsab commented on client/src/components/Header.css:
```diff
@@ -0,0 +1,46 @@
+  padding: 20px;
+}
+
+.Header__navigation__wrapper {
```
I'd make this position: absolute so it doesn't push down the content. Take a look at the pic:
![image](https://user-images.githubusercontent.com/6980777/130096126-38a881f4-7e95-4383-a3c5-c4f2c1073c01.png)

namirsab commented on client/src/components/Header.js:
```diff
@@ -1,20 +1,57 @@
-      <NavLink to="/kontakt">Kontakt</NavLink>
-      <NavLink to="/spielplatz-eintragen">Spielplatz Eintragen</NavLink>
-    </nav>
+    <header className="Header">
+      <Link to="/">
+        <img className="Header__logo" width="50px" src={logo} alt="logo" />
+      </Link>
+      <section className={toggleWrapper}>
+        <nav className={toggleNavigationLinks}>
+          <NavLink to="/">Home</NavLink>
+          <NavLink to="/map">Map</NavLink>
+          <NavLink to="/impressum">Impressum</NavLink>
+          <NavLink to="/datenschutz">Datenschutz</NavLink>
+          <NavLink to="/wir">Über Uns</NavLink>
+          <NavLink to="/kontakt">Kontakt</NavLink>
+          <NavLink to="/spielplatz-eintragen">Spielplatz Eintragen</NavLink>
+        </nav>
+        <div onClick={handleOnClick} className={toggleButton}>
+          X
+        </div>
+      </section>
+    </header>
```
Do you want to style your links? they look like default links right now. Default links are not nice. Don't be default ;)

namirsab reviewed (changes requested):
Good job @mikemangas !

I hope you had some Erfolgserlebnisse ;)

I left some comments, please take a look
